pkg/asset/tls/apiserver.go

```diff
@@ -31,27 +31,13 @@ func (a *APIServerCertKey) Dependencies() []asset.Asset {
 // Generate generates the cert/key pair based on its dependencies.
 func (a *APIServerCertKey) Generate(dependencies asset.Parents) error {
 	kubeCA := &KubeCA{}
-	installConfig := &installconfig.InstallConfig{}
-	dependencies.Get(kubeCA, installConfig)
-
-	apiServerAddress, err := cidrhost(installConfig.Config.Networking.ServiceNetwork[0].IPNet, 1)
-	if err != nil {
-		return errors.Wrap(err, "failed to get API Server address from InstallConfig")
-	}
+	dependencies.Get(kubeCA)
 
 	cfg := &CertCfg{
 		Subject:      pkix.Name{CommonName: "system:kube-apiserver", Organization: []string{"kube-master"}},
 		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
-		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 		Validity:     ValidityTenYears,
-		DNSNames: []string{
```

Comment:
We've ratcheted through changes in kas-o, mco, kubelet, kas-o, and now we can fix this.  It's never used to serve anymore